src/Infrastructure/Persistence/Configurations/DepartmentConfiguration.cs

```diff
@@ -9,6 +10,8 @@ public class DepartmentConfiguration : IEntityTypeConfiguration<Department>
     public void Configure(EntityTypeBuilder<Department> builder)
     {
         builder.HasKey(x => x.Id);
+        builder.Property(x => x.Id)
+            .HasValueGenerator<GuidValueGenerator>();
```

Comment:
should use methods other than this, as for https://github.com/dotnet/efcore/issues/28024